@here/harp-mapview-decoder/test/TileLoaderTest.ts

```diff
@@ -189,23 +188,31 @@ describe("TileLoader", function() {
                 new MockTileDecoder(),
                 0
             );
+            // This test writes an error to the console, so we disable it.
+            LoggerManager.instance.update("TileLoader", { enabled: false });
 
             dataProvider.failsRequests = true;
             const loadPromise = tileLoader.loadAndDecode();
-            // tslint:disable-next-line: no-unused-expression
             expect(loadPromise).to.not.be.undefined;
 
-            return expect(loadPromise).to.eventually.be.rejected.then(() => {
-                expect(tileLoader.state).to.equal(TileLoaderState.Failed);
-                dataProvider.failsRequests = false;
-
-                // tslint:disable-next-line: no-shadowed-variable
-                const loadPromise = tileLoader.loadAndDecode();
-                // tslint:disable-next-line: no-unused-expression
-                expect(loadPromise).to.not.be.undefined;
-
-                return expect(loadPromise).to.eventually.be.fulfilled;
-            });
+            // Chai.PromisedAssertion doesn't have .finally unfortunately.
+            return expect(loadPromise).to.eventually.be.rejected.then(
+                () => {
+                    LoggerManager.instance.update("TileLoader", { enabled: true });
+                    expect(tileLoader.state).to.equal(TileLoaderState.Failed);
+                    dataProvider.failsRequests = false;
+
+                    // tslint:disable-next-line: no-shadowed-variable
+                    const loadPromise = tileLoader.loadAndDecode();
+                    expect(loadPromise).to.not.be.undefined;
+
+                    return expect(loadPromise).to.eventually.be.fulfilled;
+                },
+                () => {
```

Comment:
This code might not be called if `expect(loadPromise).to.eventually.be.rejected` fails. See comment above.

Reply:
why not? I tested it by removing the throw new Error and returning a valid object, and it works.